kernel/src/engine/arrow_utils.rs

```diff
@@ -355,20 +358,40 @@ fn get_indices(
                                 return Err(Error::generic("map fields had more than 2 members"));
                             }
                             let inner_schema = map_type.as_struct_schema(key_name, val_name);
-                            let (parquet_advance, _children) = get_indices(
+                            let (parquet_advance, mut children) = get_indices(
                                 parquet_index + parquet_offset,
                                 &inner_schema,
                                 inner_fields,
                                 mask_indices,
                             )?;
+
                             // advance the number of parquet fields, but subtract 1 because the
                             // map will be counted by the `enumerate` call but doesn't count as
                             // an actual index.
                             parquet_offset += parquet_advance - 1;
                             // note that we found this field
                             found_fields.insert(requested_field.name());
-                            // push the child reorder on, currently no reordering for maps
-                            reorder_indices.push(ReorderIndex::identity(index));
+
+                            if children.len() != 2 {
```

Comment:
little nit: maybe could increase safety/future-proofness by getting first two elements directly? like
```rust
let [key_child, value_child] = children.try_into().map_err(...)
```

Reply:
That would be a bit safer, but we actually need to potentially modify the children, and then keep them as a vec that we put into the nested transform. So doing it this way pulls them out of the vec and then would have to put them right back in.

Since we just checked a few lines above that we have two elements, I think the safety is okay. And if arrow switches how they represent maps it'll break stuff all over, so I don't think we need to worry too much about future proofing either.